http.c

```diff
@@ -2969,17 +2970,16 @@ static void
 evhttp_send_done(struct evhttp_connection *evcon, void *arg)
 {
 	int need_close;
+	enum evhttp_cmd_type type;
 	struct evhttp_request *req = TAILQ_FIRST(&evcon->requests);
 	TAILQ_REMOVE(&evcon->requests, req, next);
 
 	if (req->on_complete_cb != NULL) {
 		req->on_complete_cb(req, req->on_complete_cb_arg);
 	}
 
-	need_close =
-	    (REQ_VERSION_BEFORE(req, 1, 1) &&
```

Comment:
If the Connection didn't set the keep-alive,the connection will be closed.

Reply:
I think this needs some more explanation, because what was written here before, seems correct. For HTTP/1.0 the default is "close", so close if there is no "keep-alive" header. For HTTP/1.1 the default is "keep-alive", so only close if there is a "close" header. You're turning `evhttp_is_connection_close()` into `!evhttp_is_connection_keepalive()`, but both are needed (and both are used) for regular http methods.

It looks like `evhttp_is_connection_close()` has an exception for `EVHTTP_PROXY_REQUEST`, and `evhttp_is_request_connection_close()` has an exception for `CONNECT` already ... so some explanation is needed for what exact request triggered the bug you are trying to fix, and why this is the best way to fix it, it is not obvious on its own.

Reply:
1.I have revised 
2. I modify the ‘assert in the ’evhttp_connection_fail_‘ function’’ when use 'http connect server mode'